doc/site/docs/02-core-concepts/04-requests.md

```diff
@@ -0,0 +1,360 @@
+  
+  print('Relative URL: $url');
+  print('Full URI: $fullUri');
+  
+  return ctx.respond(Response.ok());
+});
+```
+
+For a request to `http://localhost:8080/users/123?details=true`:
+
+- `url.path` would be relative to the handler.
+- `requestedUri` would be the complete url, including the query parameters.
+
+## Working with Query Parameters
+
+Query parameters are key-value pairs appended to the URL after a question mark (`?`). They're commonly used to pass optional data or filters to your endpoints.
+
+### Single Value Parameters
+
+Use `queryParameters` to access individual parameter values:
+
+```dart
+router.get('/search', (ctx) {
+  final query = ctx.request.url.queryParameters['query'];
+  final page = ctx.request.url.queryParameters['page'];
+  
+  if (query == null) {
+    return ctx.respond(Response.badRequest(
+      body: Body.fromString('Query parameter "query" is required'),
+    ));
+  }
+  
+  return ctx.respond(Response.ok(
+    body: Body.fromString('Searching for: $query (page: ${page ?? "1"})'),
+  ));
+});
+```
+
+For the URL `/search?query=relic&page=2`:
+
+- `query` = `"relic"`
+- `page` = `"2"`
+
+### Multiple Values
+
+Some parameters can appear multiple times in a URL. Use `queryParametersAll` to access all values:
+
+```dart
+router.get('/filter', (ctx) {
+  final tags = ctx.request.url.queryParametersAll['tag'] ?? [];
+  
+  return ctx.respond(Response.ok(
+    body: Body.fromString('Filtering by tags: ${tags.join(", ")}'),
+  ));
+});
+```
+
+For the URL `/filter?tag=dart&tag=server&tag=web`:
+
+- `tags` = `["dart", "server", "web"]`
+
+## Reading Headers
+
+HTTP headers carry metadata about the request. Relic provides type-safe access to common headers, automatically parsing them into appropriate Dart types.
+
+### Type-Safe Header Access
+
+Instead of working with raw string values, Relic's type-safe headers give you properly typed objects:
+
+```dart
+router.get('/info', (ctx) {
+  final request = ctx.request;
+  
+  // Get typed values
+  final mimeType = request.mimeType;  // MimeType? (from Content-Type)
+  final userAgent = request.headers.userAgent;  // String?
+  final contentLength = request.headers.contentLength;  // int?
+  
+  return ctx.respond(Response.ok(
+    body: Body.fromString('Browser: ${userAgent ?? "Unknown"}'),
+  ));
+});
+```
+
+### Common Headers
+
+Relic provides convenient accessors for frequently used headers:
+
+- **`mimeType`** - MIME type from Content-Type (via `request.mimeType`, not `headers`)
+- **`contentLength`** - Content length as an integer (`int?`)
+- **`userAgent`** - User agent string (`String?`)
+- **`authorization`** - Authentication credentials (`AuthorizationHeader?`)
+- **`cookie`** - Cookie header (`CookieHeader?` - use `.cookies` property for `List<Cookie>`)
+- **`host`** - Host header (`HostHeader?`)
+- **`accept`** - Content types client accepts (`AcceptHeader?`)
+- **`acceptEncoding`** - Encodings client supports (`AcceptEncodingHeader?`)
```

Comment:
This seems a bit arbitrary. What is the reason for highlighting these in particular?